src/librustc/hir/intravisit.rs

```diff
@@ -630,8 +630,8 @@ pub fn walk_path_segment<'v, V: Visitor<'v>>(visitor: &mut V,
 pub fn walk_path_parameters<'v, V: Visitor<'v>>(visitor: &mut V,
                                                 _path_span: Span,
                                                 path_parameters: &'v PathParameters) {
-    walk_list!(visitor, visit_lifetime, &path_parameters.lifetimes);
-    walk_list!(visitor, visit_ty, &path_parameters.types);
+    walk_list!(visitor, visit_lifetime, path_parameters.lifetimes());
+    walk_list!(visitor, visit_ty, path_parameters.types());
```

Comment:
The reason methods like `lifetimes` / `types` should be avoided is that this code needs to look much different (it should iterate and match over `path_parameters` - maybe with a `visit_path_param` method in `Visitor`).